crates/ruff_python_formatter/src/statement/stmt_try.rs

```diff
@@ -1,13 +1,175 @@
-use crate::{not_yet_implemented, FormatNodeRule, PyFormatter};
+use crate::comments;
+use crate::comments::{
+    dangling_node_comments, leading_comments, leading_node_comments, trailing_node_comments,
+};
+use crate::expression::parentheses::Parenthesize;
+use crate::prelude::*;
+use crate::statement::FormatRefWithRule;
+use crate::trivia::lines_before;
+use crate::{FormatNodeRule, PyFormatter};
 use ruff_formatter::{write, Buffer, FormatResult};
+use ruff_python_ast::node::AstNode;
+use ruff_python_ast::prelude::{ExceptHandler, ExceptHandlerExceptHandler, Ranged};
 use rustpython_parser::ast::StmtTry;
 
 #[derive(Default)]
 pub struct FormatStmtTry;
 
+#[derive(Copy, Clone, Default)]
+pub struct FormatExceptHandler;
+
+impl FormatRule<ExceptHandler, PyFormatContext<'_>> for FormatExceptHandler {
+    fn fmt(
+        &self,
+        item: &ExceptHandler,
+        f: &mut Formatter<PyFormatContext<'_>>,
+    ) -> FormatResult<()> {
+        let ExceptHandler::ExceptHandler(except_handler_except_handler) = item;
+        let ExceptHandlerExceptHandler {
+            range: _,
+            type_,
+            name,
+            body,
+        } = except_handler_except_handler;
+
+        let comments_info = f.context().comments().clone();
+        let leading_comments =
+            comments_info.leading_comments(except_handler_except_handler.as_any_node_ref());
+        if let [first_leading_comment, ..] = leading_comments {
+            if lines_before(
+                first_leading_comment.slice().start(),
+                f.context().contents(),
+            ) > 1
+            {
+                write!(f, [empty_line()])?;
+            }
+        }
+
+        leading_node_comments(except_handler_except_handler).fmt(f)?;
```

Comment:
Would `leading_alternate_branch_comments` work here? Could you also add a test with intentional newlines between the comments and the excepthandler to ensure we preserve the newlines correctly?

```python
try:
	pass
	
	# comment
	
# comment
except ex:
	pass
```

Reply:
well curiously your example formats correctly, but removing the blank line 3 causes the formatter to also remove the blank line 5 (i guess i'm somehow using the blank line after the. `pass` instead of the blank line after the comment. does that mean i can't use leading_alternate_branch_comments after all there? or am i doing something else wrong

Reply:
Okay I tried this locally and the issue is that we need to pass the *leading* comments of the *except handler*. 

```rust
let handler_comments = comments_info.leading_comments(handler);
write!(
    f,
    [
        leading_alternate_branch_comments(handler_comments, last_node),
        &handler.format()
    ]
)?;
```

Why not extract them from dangling comments? This is because except handlers are different from the `else` or `finally` because the except handler has a node to which we can attach the comments to. We only use dangling comments for the `else` and `finally` match cases.

Reply:
sorry after the recent refactor into `except_handler_except_handler` i'm not sure where this patch should go